core/src/apps/evolu/index_management.py

```
@@ -0,0 +1,29 @@
+    from storage.device import (
+        get_delegated_identity_key_rotation_index,
+        set_delegated_identity_key_rotation_index,
+    )
+    from trezor.messages import EvoluIndexManagementResponse
+
+    stored_index = get_delegated_identity_key_rotation_index()
+
+    if stored_index is None:
+        if msg.rotation_index is not None:
+            if msg.rotation_index < 0 or msg.rotation_index > ROTATION_INDEX_LIMIT:
+                raise ValueError(f"Rotation index must be between 0 and {ROTATION_INDEX_LIMIT}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)